datadog_checks_base/datadog_checks/base/stubs/similar.py

```diff
@@ -9,6 +9,66 @@
 MAX_SIMILAR_TO_DISPLAY = 15
 
 
+def dict_diff(expected, closest):
+    """
+    Returns an array of key/value pairs that are different between the two dicts.
+    The number of keys is fairly small so we can prioritize clarity over performance.
+    """
+    diff = []
+    for key in closest:
+        expected_value = expected.get(key)
+        closest_value = closest.get(key)
+
+        if expected_value is not None and expected_value != closest_value:
+            diff.append((key, expected_value, closest_value))
+    for key in expected:
+        if expected[key] is not None and key not in closest:
+            diff.append((key, expected[key], None))
```

Comment:
I this logically equivalent to the following? (dict keys in Python support union operations)

```python
for key in closest.keys() | expected.keys():
    expected_value = expected.get(key)
    closest_value = closest.get(key)
    if expected_value != closest_value:
        diff.append((key, expected_value, closest_value))
```

Reply:
Almost, we have to keep the None check because if expected value is None we accept any found value. Updated to the union though since it's a lot cleaner